api/apiroutes.go

```diff
@@ -1226,114 +1232,61 @@ func (c *appContext) getBlockRangeSteppedSize(w http.ResponseWriter, r *http.Req
 
 func (c *appContext) getBlockRangeSummary(w http.ResponseWriter, r *http.Request) {
 	idx0 := m.GetBlockIndex0Ctx(r)
-	if idx0 < 0 {
-		http.Error(w, http.StatusText(422), 422)
+	idx1 := m.GetBlockIndexCtx(r)
+
+	low, high := idx0, idx1
+	if idx0 > idx1 {
+		low, high = idx1, idx0
+	}
+	if low < 0 || uint32(high) > c.Status.Height() {
+		http.Error(w, "invalid block range", http.StatusBadRequest)
 		return
 	}
 
-	idx := m.GetBlockIndexCtx(r)
-	if idx < 0 {
-		http.Error(w, http.StatusText(422), 422)
+	if high-low+1 > maxBlockRangeCount {
+		http.Error(w, fmt.Sprintf("requested more than %d-block maximum", maxBlockRangeCount), http.StatusBadRequest)
 		return
 	}
 
-	// w.Header().Set("Content-Type", "application/json; charset=utf-8")
-	// N := idx - idx0 + 1
-	// summaries := make([]*apitypes.BlockDataBasic, 0, N)
-	// for i := idx0; i <= idx; i++ {
-	// 	summaries = append(summaries, c.BlockData.GetSummary(i))
-	// }
-	// writeJSON(w, summaries, c.getIndentQuery(r))
-
-	w.Header().Set("Content-Type", "application/json; charset=utf-8")
-	encoder := json.NewEncoder(w)
-	indent := c.getIndentQuery(r)
-	prefix, newline := indent, ""
-	encoder.SetIndent(prefix, indent)
-	if indent != "" {
-		newline = "\n"
-	}
-	fmt.Fprintf(w, "[%s%s", newline, prefix)
-	for i := idx0; i <= idx; i++ {
-		summary := c.DataSource.GetSummary(i)
-		if summary == nil {
-			apiLog.Debugf("Unknown block %d", i)
-			http.Error(w, fmt.Sprintf("I don't know block %d", i), http.StatusNotFound)
-			return
-		}
-		// TODO: deal with the extra newline from Encode, if needed
-		if err := encoder.Encode(summary); err != nil {
-			apiLog.Infof("JSON encode error: %v", err)
-			http.Error(w, http.StatusText(422), 422)
-			return
-		}
-		if i != idx {
-			fmt.Fprintf(w, ",%s%s", newline, prefix)
-		}
+	blocks := c.DataSource.GetSummaryRange(idx0, idx1)
```

Comment:
We need to maintain the positional relationship so that the `DESC` query can be selected in `dcrpg/queries.go` and the mod value can be set to the correct value for the skip queries.

Reply:
I meant to remove this comment after seeing you have the check in RetrieveBlockSummaryRange:
https://github.com/decred/dcrdata/pull/1590/files/762f898559d89122f05876cea15dfe4ea6e00c10#diff-87f7a577b6af45d0284e4c3fbb98ddf5R4332-R4335